### DISCUSSION TITLE
puddletag --add $file to add files to existing puddletag-GUI-session?

### DISCUSSION THREAD

**Discussion by thefoster** [puddletag/puddletag#912] [Q&A] [answered]
I'm currently (on Kubuntu/puddletag 2.0.1) using DupeGuru (little deprecated, but still best tool around) in order to deduplicate my collection.
It allows files to "open with" and i'd like to hand over those files to an existing GUI session of puddletag (--add) so that i can edit just those files that need to be altered. I basically need a cli option for drag'n'drop, where files are simply added.
I couldn't find such thing in the documentation.

**Comment by corubba** [comment]
There currently isn't such a thing. It's not undoable nor trivial, because suddenly you have to deal with communicating between different processes (IPC). The closest thing we have is specifying files/dirs to open on the command line, which is available since v2.1.0.

I don't know DupeGuru or your workflow, so I am just thinking loud here: Have "Open with" call some kind of script which stores the filename to a file. When you are ready to edit files, call a second script which reads the filenames from that file, resets the file (so it just doesn't grow endlessly), and then opens a (new) puddletag instance with the read filenames.

EDIT: Another possibility that just crossed my mind: You can also drag'n'drop files onto the puddletag window to add them. So instead of "open with" you could try dragging them directly from DupeGuru, if they support that. Or a script for use with "open with" that uses the filename to generate a synthetic drop-event in puddletag.

**Comment by thefoster** [comment]
Thanks for your answer.
I'm currently making intensive use of drag'n'drop with konqueror/dolphin, but that doesn't help to add files from another application like dupeguru that doesn't seem to support d'n'd :-/
But as you can execute a "custom command" in context of the file selection, the perfect solution (i thought) would be "puddletag --add".
But as i just discovered that that option is just for dupeguru-internal commands :-(

I have to clean up my (huge) music library from old/damaged and/or dupes, but before deleting them, i'll have to transfer the mp3-metadata to the new/undamaged files.
Dupeguru is good at finding dupes by similar filename and even content, while puddletag is great at mass tagging with very few flaws (one can ship around), so they would be a perfect team for my issue.

Maybe i'm gonna find a solution by exporting the dupelist as csv, processing some sed/awk-magic, creating symlinks in a separate dir and importing those into puddletag.

Thanks for your thoughts

**Reply by thefoster** [reply, marked as answer]
Forgot to post my final "solution" to this:
In MP3Diags you can specify 3rd-party programs, so i added puddletag as command without any options.
I now "just" have to select all files there i want to process in puddletag and right-clicking->puddletag does the basic job.
I only have to forget about the option to do this several times into the same existing puddletag-session.
Good enough for my purpose.
